src/typechecker.erl

```diff
@@ -1304,7 +1311,7 @@ expect_fun_type_union(Env, [Ty|Tys]) ->
     Any :: any,
     FieldsTy :: {fields_ty, [typed_record_field()], constraints:constraints()},
     FieldsTys :: {fields_tys, [[typed_record_field()]], constraints:constraints()},
-    TypeError :: {type_error, type()}.
+    TypeError :: {type_error, atom() | type()}.
```

Comment:
Atoms are types IUCC

Reply:
That's not the case, as above. Atom literals can be represented by `type()`, but literal atoms are not of type `type()`, see https://github.com/josefs/Gradualizer/blob/3cda2e5aa1b1313b3e02fce67986fcda3ab9a02c/src/gradualizer_type.erl#L290-L292

Reply:
Exactly :+1:

Reply:
This return value is not used so far, so maybe let's decide about what to return once we have a use for it.
https://github.com/josefs/Gradualizer/blob/3cda2e5aa1b1313b3e02fce67986fcda3ab9a02c/src/typechecker.erl#L4283

It's fine to return even just `type_error`, I just didn't want to change too much in one go.